src/Cryptol/TypeCheck/Solver/InfNat.hs

```diff
@@ -23,16 +24,33 @@ import Control.DeepSeq
 
 -- | Natural numbers with an infinity element
 data Nat' = Nat Integer | Inf
-            deriving (Show, Eq, Ord, Generic, NFData)
+            deriving (Eq, Ord, Generic, NFData)
+
+instance Show Nat' where 
+  show = \case
+    Nat n -> show n
+    Inf -> "inf"
 
 fromNat :: Nat' -> Maybe Integer
 fromNat n' =
   case n' of
     Nat i -> Just i
     _     -> Nothing
 
+instance Num Nat' where
```

Comment:
Realllllly not sure I'm a fan of this instance.  How necessary is it? `negate` and `fromInteger` are just wrong.

Reply:
The reason this was so useful to me is because I defined `Exp` to be polymorphic in its domain (which I use as either `Nat'` or `Rational`), and so then defined a bunch of operations over `Exp` that required a `Num` instance